### PR TITLE
Update citation file

### LINKED ISSUE #7477: DOC: Add a CITATION.cff file
### Issue with current documentation:

There is currently a CITATION.bib file in this repo. But, if a proper `CITATION.cff` file is set up:  
a) it could be more visible to users
b) it could provide the citation in both the APA and BibTeX versions

Example:
- video: https://www.linkedin.com/feed/update/urn:li:activity:7232762440747302912/
- sample file: https://github.com/scikit-learn/scikit-learn/blob/main/CITATION.cff
- https://citation-file-format.github.io/#/tools-for-working-with-citationcff-files
- https://github.com/citation-file-format/cffconvert

### [What is a CITATION.cff file?](https://citation-file-format.github.io/)
CITATION.cff files are plain text files with human- and machine-readable citation information for software (and datasets). Code developers can include them in their repositories to let others know how to correctly cite their software.

### Idea or request for content:

_No response_

### PR DESCRIPTION
**Pull request by Armavica**
Switch to CITATION.cff.

Validated with [`cffconvert`](https://github.com/citation-file-format/cffconvert).

## Related Issue

- [x] Closes #7477
- [ ] Related to #

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc--7479.org.readthedocs.build/en/7479/

### CODE REVIEW

**Comment by aloctavodia** [comment]
Probably, better to switch to cff, see  https://github.com/pymc-devs/pymc/issues/7477

**Comment by Armavica** [comment]
In my understanding it's for a different purpose: while the bib is for the paper, the cff is for a specific version of the software, but I could be wrong and would be happy to be corrected.

**Comment by aloctavodia** [comment]
cff is for the paper. See https://github.com/arviz-devs/preliz/blob/main/CITATION.cff as an example.

File: CITATION.bib [hunk 1]
```diff
@@ -1,10 +1,17 @@
 @article{pymc2023,
-  title={PyMC: A Modern and Comprehensive Probabilistic Programming Framework in Python},
-  author={Abril-Pla Oriol and Andreani Virgile and Carroll Colin and Dong Larry and Fonnesbeck Christopher J. and Kochurov Maxim and Kumar Ravin and Lao Jupeng and Luhmann Christian C. and Martin Osvaldo A. and Osthege Michael and Vieira Ricardo and Wiecki Thomas and Zinkov Robert},
-  journal = {{PeerJ} Computer Science},
-  publisher = {{PeerJ}},
-  volume={9},
-  pages={e1516},
-  year={2023},
-  doi={10.7717/peerj-cs.1516}
+	title = {{PyMC}: a modern, and comprehensive probabilistic programming framework in {Python}},
```
Comment (twiecki) [inline CITATION.bib]: What about the upper case?

Comment (Armavica) [reply CITATION.bib]: First of all, the published title is not in title case: https://peerj.com/articles/cs-1516/

Besides, upper or lower case are irrelevant in bibtex entries, unless wrapped in extra `{}`s. BibTeX will apply title case or normal case to all titles depending on the style sheet.

This version preserves the original case better, by wrapping `PyMC` and `Python` in extra `{}`s (see ref. 111 of https://arxiv.org/pdf/2408.09024 likely generated from the previous bib file).

**Comment by Armavica** [comment]
@aloctavodia I generated a cff file semi-manually and validated it with cffconvert.

**Comment by aloctavodia** [comment]
LGTM!

**Review by aloctavodia — APPROVED** [review]

**Pull request merged**